### PR TITLE
Added header to sheet

### PR DESCRIPTION
**Pull request by ErlingMK**
## Added / Fixed

- Added header with optional `Action` and `Cancel` buttons, and a `Title`.

## Todo List

- [x] I have tested on an Android device.
- [x] I have tested on an iOS device.
- [x] I have added unit tests

>*Please add pictures / Gifs if possible*
>*Todo List can be checked by putting a `X` inside the brackets*
>*Remember to update our `wiki` after this PR is merged*

### CODE REVIEW

File: src/Samples/DIPS.Xamarin.UI.Samples/Controls/Sheet/SheetPage.xaml.cs [hunk 1]
```diff
@@ -39,12 +39,27 @@ public class SheetPageViewModel : INotifyPropertyChanged
...
             OpenSheetCommand = new Command(() => IsSheetOpen = true);
             OnOpenCommand = new Command<string>(SheetOpened);
             OnCloseCommand = new Command<string>(SheetClosed);
+            CancelCommand = new CancelSheetCommand<string>(
+                str =>
+                {
+                    
+                },
+                str => m_cancelCanExecute, str => true);
+            ActionCommand = new Command<string>(
```
Comment (haavamoa) [inline src/Samples/DIPS.Xamarin.UI.Samples/Controls/Sheet/SheetPage.xaml.cs:57]: Remove logic and add `//Do work when action is pressed`

File: src/Samples/DIPS.Xamarin.UI.Samples/Controls/Sheet/SheetPage.xaml.cs [hunk 2]
```diff
@@ -39,12 +39,27 @@ public class SheetPageViewModel : INotifyPropertyChanged
...
         {
             OpenSheetCommand = new Command(() => IsSheetOpen = true);
             OnOpenCommand = new Command<string>(SheetOpened);
             OnCloseCommand = new Command<string>(SheetClosed);
+            CancelCommand = new CancelSheetCommand<string>(
+                str =>
+                {
+                    
+                },
+                str => m_cancelCanExecute, str => true);
```
Comment (haavamoa) [inline src/Samples/DIPS.Xamarin.UI.Samples/Controls/Sheet/SheetPage.xaml.cs:56]: Open the `canCloseSheet` lamba and add `//Do logic to determine if the sheet should close` and return `true`

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 3]
```diff
@@ -251,8 +284,230 @@ private static void BackgroundColorPropertyChanged(BindableObject bindable, obje
...
+        {
+            get => GetValue(CancelCommandParameterProperty);
+            set => SetValue(CancelCommandParameterProperty, value);
+        }
+
+        /// <summary>
+        ///     Gets or sets the command that is invoked when the cancel button is activated.
+        ///     This is a bindable property.
+        /// </summary>
+        public ICancelSheetCommand CancelCommand
```
Comment (haavamoa) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs:395]: One drawback of this being `ICancelSheetCommand` is that new consumers that don't know any better will most likely create an `ICommand` the first time they test this without reading the docs.
What happens then is that their Command won't get executed as the binding does not match. A

One kind of nice way to notify the consumer that they have set the wrong type of command is to use a `TypeConverter` and throw if the binding is not the correct type. Would you mind using a couple of minutes on that and see if it works, if it takes to much time w e an do it later.

There might be other ways to do it aswell though.

Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs:395]: One a second thought, not sure if it will work. But we can try.

Comment (ErlingMK) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs:395]: Can change to be of type `ICommand`? Close sheet if its set to a `Command`.

Comment (ErlingMK) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs:395]: And still handle if its a `CloseSheetCommand`

Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs:395]: That's interesting and I think its a good solution. Thoughts @Illedan?

File: src/Samples/DIPS.Xamarin.UI.Samples/Controls/Sheet/SheetPage.xaml [hunk 4]
```diff
@@ -15,42 +15,51 @@
...
+                                VerticalContentAlignment="{Binding VerticalContentAlignment}"
+                                ContentColor="{Binding ContentColor}"
+                                HeaderColor="{Binding HeaderColor}"
+                                HandleColor="{Binding HandleColor}"
+                                HasShadow="{Binding HasShadow}"
+                                Position="{Binding Position}"
+                                MinPosition="{Binding MinPosition}"
+                                MaxPosition="{Binding MaxPosition}"
+                                ActionCommand="{Binding ActionCommand}"
+                                ActionCommandParameter="Hei"
```
Comment (haavamoa) [inline src/Samples/DIPS.Xamarin.UI.Samples/Controls/Sheet/SheetPage.xaml:35]: No need to add these parameters, it's obvious that we have parameters 👍 You will probably have to change the commands in the code behind view model as well.

File: src/DIPS.Xamarin.UI/Internal/Xaml/SheetView.xaml [hunk 5]
```diff
@@ -31,20 +33,76 @@
             </Grid.RowDefinitions>
             <Grid x:Name="HeaderGrid"
                   Grid.Row="0"
-                  Padding="0,5,0,5"
+                  RowSpacing="0"
+                  ColumnSpacing="0"
                   BackgroundColor="{Binding HeaderColor}">
-                <BoxView HeightRequest="5"
+
+                <Grid.RowDefinitions>
```
Comment (haavamoa) [inline src/DIPS.Xamarin.UI/Internal/Xaml/SheetView.xaml:40]: You need to update the code behind `Initialize` method to reflect these changes, setting the the sheet to start from Top does not work now.

Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Internal/Xaml/SheetView.xaml:40]: So, to sum up the `Initialize`: It's whats actually is setting up the Grid, this isn't very obvious.
The reason I've done it this way is to not have to write double XAML for both cases. I totally forgot about this and it's not really obvious.

Maybe we can add a internal XAML comment here to remember ourselves this if we mess with the grid.

**Comment by haavamoa** [comment]
Remember to update wiki with new properties :)

**Pull request merged**